src/drawing/livestroke/register.ts

```diff
@@ -30,10 +31,33 @@ const register: Record<ToolType, (stroke: BoardStroke) => void> = {
         const { strokes } = board.getState().pageCollection[stroke.pageId]
         const selectedStrokes = matchStrokeCollision(
             strokes,
-            getSelectionPolygon(stroke.points)
+            getRectanglePolygon(stroke.points)
         )
         board.setTransformStrokes(Object.values(selectedStrokes), stroke.pageId)
     },
+    [ToolType.Textfield]: (stroke: BoardStroke) => {
+        const width = Math.abs(stroke.points[2] - stroke.points[0])
+        const height = Math.abs(stroke.points[2] - stroke.points[0])
+
+        if (width < TEXTFIELD_MIN_WIDTH && height < TEXTFIELD_MIN_HEIGHT) {
+            const { strokes } = board.getState().pageCollection[stroke.pageId]
+            const selectedStrokes = matchStrokeCollision(
+                strokes,
+                getRectanglePolygon(stroke.points),
+                ToolType.Textfield
+            )
```

Comment:
It should be enough and more efficient to check the collision of only a single point here, no need to calculate the collision of the rectangle polygon created with `getRectanglePolygon`. Can we adjust `matchStrokeCollision` to also accept a point instead of a polygon?

Reply:
Good idea, i'll try to implement that

Reply:
check out the new changes, i refactored everything a bit